lib/puppet/property.rb

```diff
@@ -55,6 +55,10 @@ class Puppet::Property < Puppet::Parameter
   #
   attr_writer :noop
 
+  # @!attribute [rw] sensitive
+  #   @return [true, false] If this property has been tagged as sensitive.
+  attr_accessor :sensitive
```

Comment:
I don't think so? We never sync parameters, so they shouldn't show up in reports or logs (unless the provider itself logs information taken from a parameter)